src/providers/php.rs

```diff
@@ -0,0 +1,211 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use crate::nixpacks::{
+    app::App,
+    environment::EnvironmentVariables,
+    nix::pkg::Pkg,
+    phase::{BuildPhase, InstallPhase, SetupPhase, StartPhase},
+};
+
+use super::{node::NodeProvider, Provider};
+
+pub struct PhpProvider;
+
+impl Provider for PhpProvider {
+    fn name(&self) -> &str {
+        "php"
+    }
+
+    fn detect(
+        &self,
+        app: &crate::nixpacks::app::App,
+        _env: &crate::nixpacks::environment::Environment,
+    ) -> anyhow::Result<bool> {
+        Ok(app.includes_file("composer.json") || app.includes_file("index.php"))
+    }
+
+    fn setup(
+        &self,
+        app: &crate::nixpacks::app::App,
+        env: &crate::nixpacks::environment::Environment,
+    ) -> anyhow::Result<Option<crate::nixpacks::phase::SetupPhase>> {
+        let nodejs = NodeProvider {};
+
+        let php_pkg = match self.get_php_package(app) {
+            Ok(Some(php_package)) => php_package,
+            _ => "php".to_string(),
+        };
+        let mut pkgs = vec![
+            Pkg::new(&php_pkg),
+            Pkg::new("perl"),
+            Pkg::new("nginx"),
+            Pkg::new(&format!("{}Packages.composer", &php_pkg)),
+        ];
+        if let Ok(php_extensions) = self.get_php_extensions(app) {
+            for extension in php_extensions {
+                pkgs.push(Pkg::new(&format!("{}Extensions.{}", &php_pkg, extension)));
+            }
+        }
+
+        if let Ok(true) = nodejs.detect(app, env) {
+            if let Ok(Some(mut node_setup)) = nodejs.setup(app, env) {
+                pkgs.append(&mut node_setup.pkgs);
+            }
+        }
+
+        Ok(Some(SetupPhase::new(pkgs)))
+    }
+
+    fn install(
+        &self,
+        app: &crate::nixpacks::app::App,
+        env: &crate::nixpacks::environment::Environment,
+    ) -> anyhow::Result<Option<crate::nixpacks::phase::InstallPhase>> {
+        let mut cmd = String::new();
+        cmd.push_str("mkdir -p /var/log/nginx && mkdir -p /var/cache/nginx");
+        let nodejs = NodeProvider {};
+        if app.includes_file("composer.json") {
+            cmd.push_str(" && composer install");
+        };
+        if nodejs.detect(app, env)? {
+            if !cmd.is_empty() {
+                cmd.push_str(" && ");
+            }
+            cmd.push_str(
+                nodejs
+                    .install(app, env)?
+                    .as_ref()
+                    .unwrap()
+                    .cmd
+                    .as_ref()
+                    .unwrap()
+                    .as_str(),
+            );
+        }
+        Ok(Some(InstallPhase::new(cmd)))
+    }
+
+    fn build(
+        &self,
+        app: &crate::nixpacks::app::App,
+        _env: &crate::nixpacks::environment::Environment,
```

Comment:
nit: These can be moved to imports so we don't have to reference by `crate::nixpacks` everywhere